libevmasm/RuleList.h

```diff
@@ -571,6 +571,20 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		feasibilityFunction
 	});
 
+	rules.push_back({
+		Builtins::BYTE(A, Builtins::SHL(B, X)),
+		[=]() -> Pattern { return Builtins::BYTE(A.d() + B.d() / 8, X); },
+		false,
+		[=] { return B.d() % 8 == 0 && A.d() <= 32 && B.d() <= 256; }
+	});
+
+	rules.push_back({
+		Builtins::BYTE(A, Builtins::SHR(B, X)),
+		[=]() -> Pattern { return A.d() < B.d() / 8 ? Word(0) : Builtins::BYTE(A.d() - B.d() / 8, X); },
+		false,
+		[=] { return B.d() % 8 == 0 && A.d() <= Pattern::WordSize / 8 && B.d() <= Pattern::WordSize; }
```

Comment:
```suggestion
		[=] { return B.d() % 8 == 0 && A.d() < Pattern::WordSize / 8 && B.d() <= Pattern::WordSize; }
```
For ``A >= 32`` we should have different rules anyways. And for ``A=32`` I get a counterexample (
```[A = 32,
 X = 3213876088517980551083924184682325205044405987565585670602752,
 B = 208]
```
Probably easier to think about ``X = -1U`` (all bits set) instead.
Because ``A - B/8`` here is ``6`` and ``BYTE(6, -1U)`` is not zero. ``BYTE(32, SHR(B, 208))`` is though.

Reply:
Since I just pushed the proof and have the branch here anyways, I'll just change it.